.github/workflows/bench.yml

```diff
@@ -197,14 +197,15 @@ jobs:
               echo
             } >> results.md
           fi
-          grep -Ev 'ignored|running \d+ tests|%\)' results.txt |\
-            sed -E  -e 's/(Performance has regressed.)/:broken_heart: **\1**/gi' \
-                    -e 's/(Performance has improved.)/:green_heart: **\1**/gi' \
-                    -e 's/^ +/   /gi' \
-                    -e 's/^([a-z0-9].*)$/* **\1**/gi' \
-                    -e 's/(change:[^%]*% )([^%]*%)(.*)/\1**\2**\3/gi' \
-              >> results.md
+          sed -E -e 's/^                 //gi' \
+                 -e 's/((change|time|thrpt):[^%]*% )([^%]*%)(.*)/\1<b>\3<\/b>\4/gi' results.txt |\
+            perl -p -0777 -e 's/(.*?)\n(.*?)((No change|Change within|Performance has).*?)\n(.*?)\n\n/<details><summary>$1: $3<\/summary><pre>\n$2$5<\/pre><\/details>\n/gs' |\
+            sed -E -e 's/(Performance has regressed.)/:broken_heart: <b>\1<\/b>/gi' \
+                   -e 's/(Performance has improved.)/:green_heart: <b>\1<\/b>/gi' \
+                   -e 's/^ +((<\/pre>|Found).*)/\1/gi' \
+                   -e 's/^<details>(.*Performance has.*)/<details open>\1/gi' >> results.md
```

Comment:
I feel obligated to say this: perl?  Really?

Seriously though, this might be the time that you write some python code.  It will be more verbose, but this business of matching across multiple lines is a mess.

(That git has a dependency on perl is something I find very annoying.  That language is just bad.  It might even be worse than C++ for me.)

Reply:
Yeah, I don't love perl, but it seemed quickest to just use it. It's basically just a multi-line regex engine here.